### ISSUE TITLE
Full Disk Encryption not on an NVME Drive!

### ISSUE THREAD

**Issue by Ragebone** [calamares/calamares#697]
Hi,
it seems to me that the checkbox Full disk encryption in Calamares isn't working on NVME drives.
I post this here because i assume that the Manjaro part of calamaris is just small visual and naming addition. 

Sata SSD Work flawless for me.
PCIe NVME Samsung 950 pro, won't install at all with full disk encryption.
Without encryption, it installs fine.
The passwords and settings work on a SATA drive, so it cant be something like a fu**ed password.

Crash at 20% with the following message:

`Installation Failed`
`failed at creating Filesystem on partion /dev/nvme0n1p2`
`=================`
`Command: cryptsetup -s 512 --branch-mode --force-password luksFormat /dev/nvme0n1p2`
`=================`
`Command: cryptsetup open /dev/nvme0n1p2 luks-7......`
`=================`
`Command: mkfs.ext4 -qF /dev/mapper/nvme0n1p2`

Just know that i have tried a lot over the last week.
I'm clueless, please help, much appreciated!
Thanks a lot!
Sincerely Norman

**Comment by nshaud** [comment]
Same problem with a Samsung NVMe SSD on the Lenovo Ideapad 710s, installation fails after LUKS filesystem creation.

**Comment by billwanjohi** [comment]
You might have seen, I filed this as a kpmcore bug: https://bugs.kde.org/show_bug.cgi?id=379205. A maintainer just merged a patch that should fix this, so I'll try to test later. I think that will entail installing kpmcore-git, then calamares-git, modifying the BUILDPKG to point at the manjaro/calamares fork. We'll see.

**Comment by sguberman** [comment]
Same problem here. Same setup as OP.

**Comment by philmmanjaro** [comment]
Thx for the info. I try to update **kpmcore** in Manjaro to fix this for our distro aswell.

**Comment by billwanjohi** [comment]
I tried to test this last night, but was stymied by the lack of a general-purpose calamares-git recipe in AUR. Through much trial and error, I got apricity-calamares-git to build off of kpmcore-git and manjaro/calamares source, but I wasn't able to launch.

Would be good to have some instructions on how to recompile the installer on a live boot, because I've been grappling around in the dark for the last couple of days. The ci buildall script seems like it's at least close to what I need, but I haven't been very successful.

**Comment by billwanjohi** [comment]
Ok, installing the PKGBUILD from manjaro/packages-extra/calamares was much more successful. Was able to partition and get much further along in the process. Installation still failed, on grub-install, but not obvious that that's related.

**Comment by philmmanjaro** [comment]
I've added now a new package of **kpmcore** to our unstable repos. Please test and give feedback. Regarding **grub** we have to see what that might be the issue:

```
sudo pacman -U http://repo.manjaro.org.uk/pool/overlay/kpmcore-3.0.3-1.1-$(uname -m).pkg.tar.xz
sudo -E calamares -d
```

**Comment by nshaud** [comment]
@philmmanjaro Thanks ! I managed to successfully install Manjaro using the updated **kpmcore** and **calamares** with full disk encryption on the NVMe drive. No problem with **grub** on my side.

**Comment by philmmanjaro** [comment]
@nshaud: seems we are on the right track then ;)

**Comment by billwanjohi** [comment]
OK, I was able to complete a luks-encrypted install with @philmmanjaro's two-step after manually setting the /boot mount point in the partition step, so I can also confirm resolution (for those distros that recompile kpmcore on master). I'll file separate bugs about the many bootloader problems I've now unearthed.

**Comment by dtoenies** [comment]
After applyingthe workaround I got this error message on my Lenovo T470p with an NVME and encryption.
```
<class 'subprocess.CalledProcessError'>
Command 'grub-install --target=x86_64-efi --efi-directory=/boot/efi --bootloader-id=Manjaro --force' returned non-zero exit status 1.
Traceback:
  File "/usr/lib/calamares/modules/bootloader/main.py", line 334, in run
    prepare_bootloader(fw_type)

  File "/usr/lib/calamares/modules/bootloader/main.py", line 305, in prepare_bootloader
    install_grub(efi_directory, fw_type)

  File "/usr/lib/calamares/modules/bootloader/main.py", line 236, in install_grub
    "--force"])

  File "<string>", line 2, in <module>

```

**Comment by mlitwiniuk** [comment]
@philmmanjaro I can confirm, that updating kmpcore fixes problem on X1 Carbon 5

**Comment by philmmanjaro** [comment]
OK, thx for the info. The new **kpmcore** is also included in [Manjaro 17.0.2-pre1](https://manjaro.org/manjaro-preview-releases/).
